### DISCUSSION TITLE
ALE fails to lint - says "No active LSPs"

### DISCUSSION THREAD

**Discussion by JESii** [w0rp/ale#3629] [Q&A - Ask for help with problems]
## Information

**VIM version**

VIM - Vi IMproved 8.2 (2019 Dec 12, compiled Dec  6 2020 16:07:31)
macOS version
Operating System: 
MacOS Cataling 10.15.7 (19H2)
## What went wrong

I've been using ALE for several years now and love it. Recently something broke and now ALE doesn't work in my code.

1. No errors of any kind are displayed for my code
2. ALEInfo shows no command history (see below)
3. ALECodeAction shows `No active LSPs`
4. Running `npx eslint -f unix --stdin --stdin-filename src/App.js < src/App.js` produces the expected output:
```
/Users/jseidel/ppg/formeLife/trainer-web-app/src/App.js:8:10: 'useAuth0' is defined but never used. [Error/no-unused-vars]
/Users/jseidel/ppg/formeLife/trainer-web-app/src/App.js:43:40: 'isAuthenticated' is not defined. [Error/no-undef]
/Users/jseidel/ppg/formeLife/trainer-web-app/src/App.js:45:7: 'isLoading' is not defined. [Error/no-undef]
```
5. `npx eslint --version` shows `V7.14.0`
6. `npx stylelint --version` shows `13.8.0`.
7. I use `javascript.jsx` as the filetype for both `.js` and `.jsx` files
```
  autocmd BufNewFile,BufRead *.jsx set filetype=javascript.jsx
  autocmd BufNewFile,BufRead *.js set filetype=javascript.jsx
```
I've tried `:set ft=javascript` but that doesn't fix the problem

## Reproducing the bug

This happens with any javascript file that I edit.

### :ALEInfo

```
 Current Filetype: javascript.jsx
Available Linters: ['eslint', 'fecs', 'flow', 'flow-language-server', 'jscs', 'jshint', 'standard', 'tsserver', 'xo', 'csslint', 'fecs', 'stylelint', 'eslint', 'fecs', 'flow', 'flow-language-server', 'jscs', 'jshint', 'standard', 'tsserver', 'xo']
  Enabled Linters: ['stylelint', 'eslint']
  Ignored Linters: []
 Suggested Fixers: 
  'eslint' - Apply eslint --fix to a file.
  'fecs' - Apply fecs format to a file.
  'importjs' - automatic imports for javascript
  'prettier' - Apply prettier to a file.
  'prettier_eslint', 'prettier-eslint' - Apply prettier-eslint to a file.
  'prettier_standard', 'prettier-standard' - Apply prettier-standard to a file.
  'remove_trailing_lines' - Remove all blank lines at the end of a file.
  'standard' - Fix JavaScript files using standard --fix
  'trim_whitespace' - Remove all trailing whitespace characters at the end of every line.
  'xo' - Fix JavaScript/TypeScript files using xo --fix.
 Linter Variables:

 Global Variables:

let g:ale_cache_executable_check_failures = v:null
let g:ale_change_sign_column_color = v:null
let g:ale_command_wrapper = v:null
let g:ale_completion_delay = v:null
let g:ale_completion_enabled = 0
let g:ale_completion_max_suggestions = v:null
let g:ale_disable_lsp = 0
let g:ale_echo_cursor = 1
let g:ale_echo_msg_error_str = 'Error'
let g:ale_echo_msg_format = v:null
let g:ale_echo_msg_info_str = 'Info'
let g:ale_echo_msg_warning_str = 'Warning'
let g:ale_enabled = 1
let g:ale_fix_on_save = 0
let g:ale_fixers = {'*': ['remove_trailing_lines', 'trim_whitespace'], 'javascript': ['eslint']}
let g:ale_history_enabled = 1
let g:ale_history_log_output = 1
let g:ale_keep_list_window_open = v:null
let g:ale_lint_delay = 200
let g:ale_lint_on_enter = 1
let g:ale_lint_on_filetype_changed = 1
let g:ale_lint_on_insert_leave = 1
let g:ale_lint_on_save = 1
let g:ale_lint_on_text_changed = 'normal'
let g:ale_linter_aliases = {'jsx': ['css', 'javascript']}
let g:ale_linters = {'jsx': ['stylelint', 'eslint'], 'javascript.jsx': ['stylelint', 'eslint'], 'bash': ['sh'], 'javascript': ['stylelint', 'eslint']}
let g:ale_linters_explicit = 1
let g:ale_linters_ignore = {}
let g:ale_list_vertical = v:null
let g:ale_list_window_size = v:null
let g:ale_loclist_msg_format = v:null
let g:ale_lsp_root = {}
let g:ale_max_buffer_history_size = 20
let g:ale_max_signs = v:null
let g:ale_maximum_file_size = v:null
let g:ale_open_list = v:null
let g:ale_pattern_options = v:null
let g:ale_pattern_options_enabled = v:null
let g:ale_set_balloons = 0
let g:ale_set_highlights = 1
let g:ale_set_loclist = 1
let g:ale_set_quickfix = 0
let g:ale_set_signs = 1
let g:ale_sign_column_always = 1
let g:ale_sign_error = v:null
let g:ale_sign_info = v:null
let g:ale_sign_offset = v:null
let g:ale_sign_style_error = v:null
let g:ale_sign_style_warning = v:null
let g:ale_sign_warning = v:null
let g:ale_sign_highlight_linenrs = v:null
let g:ale_statusline_format = v:null
let g:ale_type_map = v:null
let g:ale_use_global_executables = v:null
let g:ale_virtualtext_cursor = 0
let g:ale_warn_about_trailing_blank_lines = 1
let g:ale_warn_about_trailing_whitespace = 1
  Command History:
```
**NOTE: No command history**

**Comment by JESii** [comment]
OMG: I tried a number of things over a period of serval hours (ALE is fundamental to my work flow) none of which seemed to help:

1. Reinstall ALE
1. Reinstall stylelint, eslint
1. Try prettier
1. Making many changes to the configuration/settings, including stripping all but the most fundamental pugins.
1. rebooting my machine

This morning, I tried installing an LSP, just for grins: typescript-language-server. This didn't help, but when I uninstalled it and tried ALEInfo again, now it's working!
(executable check - success) /Users/jseidel/dev/reactjs/React-the-ES6-Way/node_modules/.bin/stylelint
(finished - exit code 78) ['bash', '-c', '''/Users/jseidel/dev/reactjs/React-the-ES6-Way/node_modules/.bin/stylelint''  --stdin-filename ''/Users/jseidel/dev/reactjs/React-the-ES6-Way/.eslin
trc.js'' < ''/var/folders/xh/31s8mxjs6ts3vtlzg8qh17nw0000gn/T/vGJpOs0/2/.eslintrc.js''']
<<<NO OUTPUT RETURNED>>>
(executable check - success) /Users/jseidel/dev/reactjs/React-the-ES6-Way/node_modules/eslint/bin/eslint.js
(finished - exit code 0) ['bash', '-c', 'cd ''/Users/jseidel/dev/reactjs/React-the-ES6-Way'' && ''/Users/jseidel/dev/reactjs/React-the-ES6-Way/node_modules/eslint/bin/eslint.js'' -f json --s
tdin --stdin-filename ''/Users/jseidel/dev/reactjs/React-the-ES6-Way/.eslintrc.js'' < ''/var/folders/xh/31s8mxjs6ts3vtlzg8qh17nw0000gn/T/vGJpOs0/3/.eslintrc.js''']
<<<OUTPUT STARTS>>>
[{"filePath":"/Users/jseidel/dev/reactjs/React-the-ES6-Way/.eslintrc.js","messages":[],"errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0,"usedDeprecatedRules":[{"
ruleId":"lines-around-directive","replacedBy":["padding-line-between-statements"]},{"ruleId":"no-spaced-func","replacedBy":["func-call-spacing"]},{"ruleId":"global-require","replacedBy":[]},
{"ruleId":"no-buffer-constructor","replacedBy":[]},{"ruleId":"no-new-require","replacedBy":[]},{"ruleId":"no-path-concat","replacedBy":[]}]}]
<<<OUTPUT ENDS>>>
The stylelint issue is because there's no CSS in this simple file. And eslint is working as well in my development  machine. I'll do a little more testing and then close this issue when I'm sure things are stable.
Thanks again for an awesome plugin!

**Comment by JESii** [comment]
Would love to have some understanding of what happened here, but I'm overjoyed to have found a solution.

**Comment by JESii** [comment]
Well... just happened to me again. Only thing I can think of is that I installed `prettier` for a project and now ALE doesn't run anything, even though it appears to be properly configured as it was.
Exactly same problem as before; ALEInfo show no command history (and therefore no linting errors) while 
`npx eslint -f unix --stdin --stdin-filename src/App.js < src/App.js `
displays the lining errors as expected.

What is keeping ALE from actually running the required commands? Anything I can do to help debug?

And now its working on most files but still failing on the src/App.js file... weird.

**Comment by w0rp** [comment]
Look at the `Command History:` lines and see what's being run. It will at least tell you that it checked to see if an executable exists or not and failed to find one to run. Check if `node_modules/.bin/eslint` exists in the directory or some ancestor directory.

**Comment by JESii** [comment]
Thanks, @w0rp -- In those cases where it fails, there is literally no `Command History`
```
let g:ale_warn_about_trailing_whitespace = 1
  Command History:
<one blank line>
```

**Comment by JESii** [comment]
And yes, `node_modules/.bin/eslint` does exist

**Comment by w0rp** [comment]
If it's not even checking if any executables that exist, I have no idea. I use ALE every day at work with ESLint, TypesScript, and Pyright, so it's not something wrong with ALE. You'll have to eliminate one issue at a time like searching for any copies of ALE in runtimepath, removing them all and reinstalling the plugin, looking for anything in your Vim settings that might cause an issue. It could be anything.

**Comment by JESii** [comment]
Yeah; really weird.. it seems to come and go. Thanks; I’ll try looking for multiple copies of ALE and also reinstalling.

An d yes, I use it every day on multiple projects, have for years, and love it. Just wish I could figure out what’s going on.

Cheers…jon
Stay safe; be well.

**Reply by w0rp** [reply]
I've converted the Issue to a fancy new "Discussion" now so other people can suggest answers. It's like Stack Overflow, but on GitHub.

**Reply by w0rp** [reply]
I might not convert other historic issues. I just wanted to try this feature out.
